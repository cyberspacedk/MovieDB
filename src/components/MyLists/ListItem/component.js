import React from 'react';
import PropTypes from 'prop-types';
import { Col, Icon, Card, Modal, Typography } from 'antd';

const { Paragraph, Title } = Typography;

const ListItem = ({ item, handleListDetails, deleteListRequest }) => {
  const showDeleteModal = e => {
    e.stopPropagation();
    Modal.confirm({
      title: 'Do you want to delete this list?',
      onOk() {
        deleteListRequest(item.id);
      },
      onCancel() {},
    });
  };

  return (
    <Col
      className="my-list-card"
      xs={{ span: 12 }}
      sm={{ span: 8 }}
      mg={{ span: 6 }}
      lg={{ span: 4 }}
      xl={{ span: 4 }}
    >
      <Card
        hoverable
        className="top-margin"
        actions={[
          <Icon key="delete" type="delete" onClick={showDeleteModal} />,
        ]}
        onClick={handleListDetails}
      >
        <Title level={4}>{item.name}</Title>
        <Paragraph level={4}>{item.description}</Paragraph>
      </Card>
    </Col>
  );
};

ListItem.propTypes = {
  item: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleListDetails: PropTypes.func.isRequired,
  deleteListRequest: PropTypes.func.isRequired,
};

export default ListItem;
