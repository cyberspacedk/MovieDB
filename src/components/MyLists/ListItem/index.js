import ListItemContainer from './container';

export default ListItemContainer;
