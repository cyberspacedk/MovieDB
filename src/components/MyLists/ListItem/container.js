import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'antd';
import ListItem from './component';

class ListItemContainer extends Component {
  handleListDetails = () => {
    const { history, item } = this.props;
    history.push(`/lists/${item.id}`);
  };

  onOk = () => {
    const {
      deleteListRequest,
      item: { id },
    } = this.props;
    deleteListRequest(id);
  };

  handleDeleteModal = e => {
    e.stopPropagation();
    Modal.confirm({
      title: 'Do you want to delete this list?',
      onOk: this.onOk,
    });
  };

  render() {
    const { item } = this.props;
    return (
      <ListItem
        item={item}
        handleListDetails={this.handleListDetails}
        handleDeleteModal={this.handleDeleteModal}
      />
    );
  }
}

ListItemContainer.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
  deleteListRequest: PropTypes.func.isRequired,
  item: PropTypes.shape({
    id: PropTypes.number,
  }).isRequired,
};

export default ListItemContainer;
