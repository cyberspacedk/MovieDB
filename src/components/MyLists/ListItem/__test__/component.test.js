import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../component';

describe('Component: ListItem', () => {
  const props = {
    handleListDetails: jest.fn(),
    handleDeleteModal: jest.fn(),
    item: {
      name: 'Comedy',
      description: 'List Description',
    },
  };

  const container = shallow(<ListItem {...props} />);

  it('Snapshot: should match', () => {
    expect(container).toMatchSnapshot();
  });
});
