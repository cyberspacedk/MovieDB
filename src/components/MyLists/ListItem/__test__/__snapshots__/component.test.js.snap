// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ListItem Snapshot: should match 1`] = `
<Col
  className="my-list-card"
  lg={
    Object {
      "span": 4,
    }
  }
  mg={
    Object {
      "span": 6,
    }
  }
  sm={
    Object {
      "span": 8,
    }
  }
  xl={
    Object {
      "span": 4,
    }
  }
  xs={
    Object {
      "span": 12,
    }
  }
>
  <Card
    actions={
      Array [
        <Icon
          onClick={[MockFunction]}
          type="delete"
        />,
      ]
    }
    className="top-margin"
    hoverable={true}
    onClick={[MockFunction]}
  >
    <Title
      level={4}
    >
      Comedy
    </Title>
    <Paragraph
      level={4}
    >
      List Description
    </Paragraph>
  </Card>
</Col>
`;
